crates/router/src/core/payment_methods/cards.rs

```diff
@@ -139,74 +149,158 @@ pub async fn update_customer_payment_method(
 }
 
 #[instrument(skip_all)]
-pub async fn add_card(
+pub async fn add_card_hs(
     state: &routes::AppState,
     req: api::CreatePaymentMethod,
     card: api::CardDetail,
     customer_id: String,
     merchant_account: &storage::MerchantAccount,
 ) -> errors::CustomResult<api::PaymentMethodResponse, errors::VaultError> {
     let locker = &state.conf.locker;
+    let jwekey = &state.conf.jwekey;
     let db = &*state.store;
     let merchant_id = &merchant_account.merchant_id;
+
     let locker_id = merchant_account
         .locker_id
         .to_owned()
         .get_required_value("locker_id")
         .change_context(errors::VaultError::SaveCardFailed)?;
 
-    let request = payment_methods::mk_add_card_request(
+    let request = payment_methods::mk_add_card_request_hs(
+        jwekey,
         locker,
         &card,
         &customer_id,
         &req,
         &locker_id,
         merchant_id,
-    )?;
-
-    let response = if !locker.mock_locker {
-        let response = services::call_connector_api(state, request)
-            .await
-            .change_context(errors::VaultError::SaveCardFailed)?;
-
-        let response: payment_methods::AddCardResponse = match response {
-            Ok(card) => card
+    )
+    .await?;
+    let response = services::call_connector_api(state, request)
+        .await
+        .change_context(errors::VaultError::SaveCardFailed)?;
+    let stored_card_response = match response {
+        Ok(card) => {
+            let jwe_body: services::JweBody = card
                 .response
```

Comment:
What happened to `mock_locker`? are we doing this switch somewhere else?

Reply:
yes added